recipes/powershell4.rb

```diff
@@ -30,14 +30,25 @@
     # Ensure .NET 4.5 is installed or installation will fail silently per Microsoft. Only necessary for Windows 2008R2 or 7.
     include_recipe 'ms_dotnet45' if windows_version.windows_server_2008_r2? || windows_version.windows_7?
 
-    # Reboot if user specifies immediate_reboot
-    include_recipe 'powershell::windows_reboot' if node['powershell']['installation_reboot_mode'] == 'immediate_reboot'
+    # Get powershell version
+    powershell_version_cmd = 'powershell.exe $PSVersionTable.PSVersion.Major'
+    shell_out = Mixlib::ShellOut.new(powershell_version_cmd)
+    shell_out.run_command
+    powershell_version = shell_out.stdout.to_f
+
+    # Reboot if user specifies doesn't specify no_reboot and powershell version < 4
+    include_recipe 'powershell::windows_reboot' unless node['powershell']['installation_reboot_mode'] == 'no_reboot' && powershell_version < 4
```

Comment:
@tjnicholas , thanks! Code looks neat. We should just add powershell version check in other recipes as well i.e. in powershell2, powershell3 and powershell5.
This is because we shouldn't let the reboot happen in case powershell4 recipe is run and powershell 4 is already installed on the machine. Same should be the behavior of other powershell recipes too.